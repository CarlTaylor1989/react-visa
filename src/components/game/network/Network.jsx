import React from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';
import ReactRouterPropTypes from 'react-router-prop-types';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import NetworkMap from './NetworkMap';
import ProductPage from './products/ProductPage';
import { networkMapPath, networkProductPath } from '../../../config/navigation';

import './Network.scss';
import '../../generic/PageTransitions.scss';

const Network = ({ location }) => (
  <TransitionGroup className="network-transition-group">
    <CSSTransition
      key={location.pathname.replace('/', '_')}
      timeout={{ enter: 1000, exit: 1000 }}
      classNames="pageFade"
    >
      <div className="network">
        <Switch location={location}>
          <Route path={networkMapPath} component={NetworkMap} />
          <Route path={networkProductPath} component={ProductPage} />
          <Redirect to={networkMapPath} />
        </Switch>
      </div>
    </CSSTransition>
  </TransitionGroup>
);

Network.propTypes = {
  location: ReactRouterPropTypes.location.isRequired
};

export default Network;
