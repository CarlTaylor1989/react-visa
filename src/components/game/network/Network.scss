.network-transition-group {
  height: 85%;
}

.network {
  height: 100%;
  position: relative;
  z-index: 10;
}
