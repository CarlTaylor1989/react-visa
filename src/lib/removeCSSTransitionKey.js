export default function removeCSSTransitionKey(json) {
  if (json.type === 'CSSTransition') {
    return {
      ...json,
      props: {
        ...json.props,
        key: ''
      }
    };
  }
  return json;
}
