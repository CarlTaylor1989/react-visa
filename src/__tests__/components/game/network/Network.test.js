/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import Network from '../../../../components/game/network/Network';
import removeCSSTransitionKey from '../../../../lib/removeCSSTransitionKey';

let mockProps;
beforeEach(() => {
  mockProps = {
    location: {
      hash: '',
      key: '',
      pathname: '',
      search: ''
    }
  };
});

test('renders the Network component', () => {
  const wrapper = shallow(<Network {...mockProps} />);
  expect(
    toJson(wrapper, {
      noKey: false,
      map: wrapper.exists('CSSTransition') && removeCSSTransitionKey
    })
  ).toMatchSnapshot();
});
