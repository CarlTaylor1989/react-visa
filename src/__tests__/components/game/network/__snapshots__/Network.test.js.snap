// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Network component 1`] = `
<TransitionGroup
  childFactory={[Function]}
  className="network-transition-group"
  component="div"
>
  <CSSTransition
    classNames="pageFade"
    key=""
    timeout={
      Object {
        "enter": 1000,
        "exit": 1000,
      }
    }
  >
    <div
      className="network"
    >
      <Switch
        location={
          Object {
            "hash": "",
            "key": "",
            "pathname": "",
            "search": "",
          }
        }
      >
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(NetworkMap)",
              "type": [Function],
            }
          }
          path="/game/network/map"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(ProductPage)",
              "type": [Function],
            }
          }
          path="/game/network/product"
        />
        <Redirect
          to="/game/network/map"
        />
      </Switch>
    </div>
  </CSSTransition>
</TransitionGroup>
`;
